web/toolbar.js

```diff
@@ -0,0 +1,276 @@
+      this.pageNumber = 0;
+      this.pageLabel = null;
+      this.hasPageLabels = false;
+      this.pagesCount = 0;
+      this.pageScaleValue = DEFAULT_SCALE_VALUE;
+      this.pageScale = null;
+      this._updateUIState(true);
+    },
+
+    _bindListeners: function Toolbar_bindClickListeners() {
+      var eventBus = this.eventBus;
+
+      this.items.previous.addEventListener('click', function() {
+        eventBus.dispatch('nextpage');
+      });
+
+      this.items.next.addEventListener('click', function() {
+        eventBus.dispatch('previouspage');
+      });
+
+      this.items.zoomIn.addEventListener('click', function() {
+        eventBus.dispatch('zoomin');
+      });
+
+      this.items.zoomOut.addEventListener('click', function() {
+        eventBus.dispatch('zoomout');
+      });
+
+      this.items.pageNumber.addEventListener('click', function() {
+        this.select();
+      });
+
+      this.items.pageNumber.addEventListener('change', function() {
+        eventBus.dispatch('pagenumberchanged', {value: this.value});
+      });
+
+      this.items.scaleSelect.addEventListener('change', function() {
+        if (this.value === 'custom') {
+          return;
+        }
+        eventBus.dispatch('scalechanged', {value: this.value});
+      });
+
+      this.items.presentationModeButton.addEventListener('click',
+          function (e) {
+        eventBus.dispatch('presentationmode');
+      });
+
+      this.items.openFile.addEventListener('click', function (e) {
+        eventBus.dispatch('openfile');
+      });
+
+      this.items.print.addEventListener('click', function (e) {
+        eventBus.dispatch('print');
+      });
+
+      this.items.download.addEventListener('click', function (e) {
+        eventBus.dispatch('download');
+      });
+
+      // Suppress context menus for some controls
+      this.items.scaleSelect.oncontextmenu = noContextMenuHandler;
+
+      eventBus.on('localized', this._adjustScaleWidth.bind(this));
+    },
+
+    _updateUIState: function Toolbar_updateUIState(resetNumPages) {
+      function selectScaleOption(value, scale) {
+        var options = items.scaleSelect.options;
+        var predefinedValueFound = false;
+        for (var i = 0, ii = options.length; i < ii; i++) {
+          var option = options[i];
+          if (option.value !== value) {
+            option.selected = false;
+            continue;
+          }
+          option.selected = true;
+          predefinedValueFound = true;
+        }
+        if (!predefinedValueFound) {
+          var customScale = Math.round(scale * 10000) / 100;
+          items.customScaleOption.textContent =
+            mozL10n.get('page_scale_percent', {scale: customScale},
+                        '{{scale}}%');
+          items.customScaleOption.selected = true;
+        }
+      }
+
+      var pageNumber = this.pageNumber;
+      var scaleValue = (this.pageScaleValue || this.pageScale ||
+        DEFAULT_SCALE_VALUE).toString();
```

Comment:
Nit: I'm not sure that you need the `|| DEFAULT_SCALE_VALUE` part here any more, since `this.pageScaleValue` now has that as the default.